File: src/main/java/to/joeli/jass/client/strategy/mcts/TrumpfMove.java
package to.joeli.jass.client.strategy.mcts;


import to.joeli.jass.client.game.Player;
import to.joeli.jass.client.strategy.mcts.src.Move;
import to.joeli.jass.game.mode.Mode;

import java.util.Objects;

/**
 * Created by joelniklaus on 06.05.17.
 */
public class TrumpfMove implements Move {

	private final Player player;
	private final Mode chosenTrumpf;

	public TrumpfMove(Player player, Mode chosenTrumpf) {
		this.player = player;
		this.chosenTrumpf = chosenTrumpf;
	}

	public Player getPlayer() {
		return player;
	}

	public Mode getChosenTrumpf() {
		return chosenTrumpf;
	}

	@Override
	public int compareTo(Move o) {
		return 0;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		TrumpfMove that = (TrumpfMove) o;
		return Objects.equals(player, that.player) &&
				Objects.equals(chosenTrumpf, that.chosenTrumpf);
	}

	@Override
	public int hashCode() {
		return chosenTrumpf.hashCode();
	}

	@Override
	public String toString() {
		return getChosenTrumpf().toString();
	}
}

File: src/main/java/to/joeli/jass/client/strategy/mcts/TrumpfMove.kt
package to.joeli.jass.client.strategy.mcts


import to.joeli.jass.client.game.Player
import to.joeli.jass.client.strategy.mcts.src.Move
import to.joeli.jass.game.mode.Mode

/**
 * Created by joelniklaus on 06.05.17.
 */
class TrumpfMove(val player: Player, val chosenTrumpf: Mode) : Move {

    override fun compareTo(other: Move): Int {
        return 0
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as TrumpfMove?
        return player == that!!.player && chosenTrumpf == that.chosenTrumpf
    }

    override fun hashCode(): Int {
        return chosenTrumpf.hashCode()
    }

    override fun toString(): String {
        return chosenTrumpf.toString()
    }
}
